lib/ClangImporter/ImportDecl.cpp

```diff
@@ -7024,11 +7031,88 @@ ClangImporter::Implementation::createConstant(Identifier name, DeclContext *dc,
                                               Type type, StringRef value,
                                               ConstantConvertKind convertKind,
                                               bool isStatic,
+                                              bool isFunction,
                                               ClangNode ClangN) {
   auto expr = new (SwiftContext) StringLiteralExpr(value, SourceRange());
-  return createConstant(name, dc, type, expr, convertKind, isStatic, ClangN);
+  if (isFunction)
+    return createFunction(name, dc, type, expr, convertKind, isStatic, ClangN);
+  else
+    return createConstant(name, dc, type, expr, convertKind, isStatic, ClangN);
 }
 
+ValueDecl *
+ClangImporter::Implementation::createFunction(Identifier name, DeclContext *dc,
```

Comment:
Since this code is pulled from `ClangImporter::Implementation::createConstant`, could you find a way to factor its body to call this?